extensions/vertx/deployment/src/main/java/io/quarkus/vertx/deployment/EventBusConsumer.java

```diff
@@ -175,10 +195,20 @@ private static void implementInvoke(BeanInfo bean, MethodInfo method, ClassCreat
         } else {
             // Parameter is payload
             ResultHandle bodyHandle = invoke.invokeInterfaceMethod(MESSAGE_BODY, messageHandle);
-            ResultHandle returnHandle = invoke.invokeVirtualMethod(
-                    MethodDescriptor.ofMethod(bean.getImplClazz().name().toString(), method.name(),
-                            method.returnType().name().toString(), paramType.name().toString()),
-                    beanInstanceHandle, bodyHandle);
+            ResultHandle returnHandle;
+            if (headerType.isPresent()) {
+                // TODO: Handle headers from both mutiny and non-mutiny? Will it always be mutiny?
+                ResultHandle headerHandle = invoke.invokeInterfaceMethod(MESSAGE_HEADERS, messageHandle);
```

Comment:
Presumably this should handle both Vertx and Mutiny `MultiMap`.

Reply:
Yes, you'll need to wrap the original `MultiMap` with `io.vertx.mutiny.core.MultiMap.MultiMap(MultiMap)`.

@cescoffier What's the point of `io.vertx.mutiny.core.MultiMap` anyway? I mean it does not even use any mutiny type...

Reply:
@mkouba This is a class generated from the Vert.x MultiMap (because that class a the @VertxGen annotation - see https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/MultiMap.java).

Reply:
> This is a class generated from the Vert.x MultiMap

I know but I was just wondering why is a similar class needed anyway?